### PR TITLE
Gracefully reject unsupported protocol versions on websocket adapter

### PR DESCRIPTION
**Pull request by hpihkala**
With this change, `WebsocketServer` gracefully rejects requests that:
- don't contain the required version negotiation parameters `controlLayerVersion` and `messageLayerVersion`
- contain them but the given versions are not supported

The requests to open the connection (upgrade requests) are rejected with http status code `400` and an error body detailing the problem. 

It seems that many websocket clients don't show the response body to the user, potentially hiding the error message. IMO this approach is still much better than the alternative of _accepting_ the connection, sending a message, then closing it.

### CODE REVIEW

File: test/integration/broker.test.js [hunk 1]
```diff
@@ -17,29 +17,74 @@ const networkPort2 = 12362
...
+describe('websocket server', () => {
+    let ws
+    let broker
+
+    afterEach(async () => {
+        if (ws) {
             ws.terminate()
-            await broker.close()
+        }
+        broker.close()
```
Comment (mirpo) [inline test/integration/broker.test.js:28]: `await` is missing

File: test/integration/broker.test.js [hunk 2]
```diff
@@ -17,29 +17,74 @@ const networkPort2 = 12362
...
         ws.on('open', async () => {
-            ws.terminate()
-            await broker.close()
             done()
         })
         ws.on('error', (err) => done(err))
     })
+
+    describe('rejections', () => {
+        beforeEach(async () => {
```
Comment (mirpo) [inline test/integration/broker.test.js:52]: It's easier to follow logic of test when `beforeEach` `afterEach` are near each other.

File: test/integration/unsupported-controllayer-version.test.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+const WebSocket = require('ws')
+const { startTracker } = require('streamr-network')
+const { wait, waitForCondition } = require('streamr-test-utils')
+
+const { startBroker } = require('../utils')
+
+const httpPort1 = 12341
+const wsPort1 = 12351
+const networkPort1 = 12361
+const trackerPort = 12371
```
Comment (mirpo) [inline test/integration/unsupported-controllayer-version.test.js:10]: If tests will run parallel, binding one port by two `servers` will fail.

File: test/integration/unsupported-controllayer-version.test.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            closed = true
+        })
+        await waitForCondition(() => gotError && closed)
+    }
+
+    it('rejects connections without preferred versions given as query parameters', async () => {
+        await testRejection(`ws://127.0.0.1:${wsPort1}/api/v1/ws`)
+    })
+
+    it('rejects connections with unsupported version parameters', async () => {
```
Comment (mirpo) [inline test/integration/unsupported-controllayer-version.test.js:50]: This test looks like partly duplicated with `test/integration/broker.test.js`

Comment (hpihkala) [reply test/integration/unsupported-controllayer-version.test.js:50]: Forgot to delete this file, apologies.

File: test/integration/broker.test.js [hunk 5]
```diff
@@ -17,29 +17,74 @@ const networkPort2 = 12362
...
         })
         ws.on('open', async () => {
-            ws.terminate()
-            await broker.close()
             done()
         })
         ws.on('error', (err) => done(err))
     })
+
+    describe('rejections', () => {
```
[the remark this answers is not in the document]
Comment (hpihkala) [reply test/integration/broker.test.js:51]: Yeah, I first wrote the tests in a separate file but then copied them here as the file already had similar connection tests from before. Forgot to delete the separate file, apologies.

**Comment by hpihkala** [comment]
Tests pass locally and on Travis, but on Travis I encountered some flakiness in unrelated test cases (in `broker.test.js` and `Storage.test.js`). Also it segfaults on Travis at the end of the test run, as is common nowadays.

**Pull request merged**